### PR TITLE
Make Policy field optional

### PR DESCRIPTION
**Pull request by Benjmhart**

### CODE REVIEW

**Comment by gitmachtl** [comment]
Whats the reason to make the policy field optional, and not be able to verify the signing?

**Comment by KtorZ** [comment]
@gitmachtl the reason is Plutus.

Monetary policy can be Plutus scripts, and when they are, there's no longer such a strict relationship with key owners. So, the verification will be done differently, and actually be _more decentralized_.

I'll submit a CIP soon on this topic, but in short: the verification will be done client side by the end-user's wallet, using a set of trusted keys managed by the user. Wallets will be shipped with a pre-registered set of well-known keys and users will have the ability to add new ones.

**Comment by gitmachtl** [comment]
thank for clarification. so this does not affect "normal" entries with a "simple-script" policy. it will still work like it is right now, correct?

**Comment by KtorZ** [comment]
Yes indeed. If the `policy` field is present, then it'll be used for verification. If not, then the verification will be deferred to the client + trusted keys.

File: metadata-lib/src/Cardano/Metadata/Transform/Reader.hs [hunk 1]
```diff
@@ -11,44 +12,30 @@ module Cardano.Metadata.Transform.Reader
...
-instance Monad f => Monad (Transform r f) where
-  (Transform fa) >>= fmb =
-    Transform $ fa >>= (ReaderT . apply . fmb)
-
-instance Alternative f => Alternative (Transform r f) where
-  empty = Transform empty
-  (Transform f1) <|> (Transform f2) = Transform $ f1 <|> f2
+newtype Transform r f a = Transform (r -> f a)
+  deriving (Functor, Applicative, Alternative, Monad)
+  via ReaderT r f
```
Comment (sevanspowell) [inline metadata-lib/src/Cardano/Metadata/Transform/Reader.hs:22]: Guess how I could tell icelandjack was involved. This is great, thank you!

File: token-metadata-creator/README.md [hunk 2]
```diff
@@ -62,6 +64,7 @@ token-metadata-creator entry 19309eb9c066253cede617dc635223ace320ae0bbdd5bd19684
 
 | Field    | Details                                                               | Command          |
 | ---      | ---                                                                   | ---              |
+| `policy` | the exact script which hashes to the `policyId`                       | `--policy \| -p` |
```
Comment (sevanspowell) [inline token-metadata-creator/README.md:67]: Yep, this is correct and should **not** be changed to match (https://github.com/cardano-foundation/cardano-token-registry/pull/296#discussion_r687291744).

The meaning of the registry entry is "The base16-encoded CBOR representation of the monetary policy script", but the description of the thing you should pass to `token-metadata-creator` is "the exact script which hashes to the policyId".

**Review by sevanspowell — APPROVED** [review]
Looks good!

**Pull request merged**